### PR TITLE
Update site to use vite + vue3

### PR DESCRIPTION
**Pull request by hcorson-dosch-usgs**
This large PR updates the site to use `vite` and `vue3`, working off of our `vue3` template repo. Unfortunately, there isn't an easy way to piecemeal these changes, so this completes the migration in one go, with many changed files. Note that I did not try to alter the code to meet the new template's best practices, particularly those relating to css, so the css remains scattered through different components, and does not reference a central `base.css` file. The site should look identical to the currently live site, but with our latest iterations of the USWDS header, USGS header, pre-footer code links, and USGS footer.

### To test
Pull changes locally, run `npm install` and then run `npm run dev`

### CODE REVIEW

File: src/components/MapImageSlider.vue [hunk 1]
```diff
@@ -1813,128 +1813,123 @@
...
-          this.svg = document.querySelector("#cartogram-svg");
-          this.pt = this.svg.createSVGPoint();
-      },
-      methods: {
-      
-        cursorPoint(evt) {
-          this.pt.x = evt.clientX; this.pt.y = evt.clientY;
-          return this.pt.matrixTransform(this.svg.getScreenCTM().inverse());
-        },
+  import GeorgiaInsetMap from '@/components/GeorgiaInsetMap.vue';
```
Comment (cnell-usgs) [inline src/components/MapImageSlider.vue:1820]:
```suggestion
  import GeorgiaInsetMap from '@/components/georgiaInsetMap.vue';
```

Comment (cnell-usgs) [reply src/components/MapImageSlider.vue:1820]: needed to get page to build

File: CHANGELOG.md
Comment (cnell-usgs) [inline CHANGELOG.md:1]: Let's keep this and add this as a major version update

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,14 +0,0 @@
...
-
-## [0.0.1] - 2020-08-14
-- Added pre-footer links to previous visualizations
-- Updated SVG Years
-- Basic Context Slider Functionality
-- Created Components out of the sections in the viz
-- Changed favicon to USGS
-- expanded data definition for title attributes in Visualization.vue and HeaderUSWDSSelection.vue
-
-[Unreleased]: https://github.com/usgs-makerspace/wbeep-viz/compare/v0.0.1...master
```
Comment (cnell-usgs) [inline CHANGELOG.md:14]: This line seems like it's for a different repo tho

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,14 +0,0 @@
...
-
-## [Unreleased]
-
-## [0.0.1] - 2020-08-14
-- Added pre-footer links to previous visualizations
-- Updated SVG Years
-- Basic Context Slider Functionality
-- Created Components out of the sections in the viz
-- Changed favicon to USGS
-- expanded data definition for title attributes in Visualization.vue and HeaderUSWDSSelection.vue
```
Comment (cnell-usgs) [inline CHANGELOG.md:12]: ## [1.1.0] - 2024-07-16
- Vue dependency updated to v3, with behind the scenes changes to adopt composition API

File: README.md [hunk 4]
```diff
@@ -1,41 +1,15 @@
 # Gages Through the Ages Visualization
 ![alt text](./markDownImages/gages.png "The SVG map from the Gages Through the Ages Visualization")
 ## The Purpose
-The Gages Through the Ages Visualization presents the viewer with a brief history of the U.S. 
-Geological Survey's (USGS) streamgaging efforts from the placement of the first USGS gage
-at Embudo, New Mexico in 1889 to the present day.
+The Gages Through the Ages Visualization presents the viewer with a brief history of the U.S. Geological Survey's (USGS) streamgaging efforts from the placement of the first USGS gage at Embudo, New Mexico in 1889 to the present day.
```
Comment (cnell-usgs) [inline README.md:4]:
```suggestion
The Gages Through the Ages Visualization presents the viewer with a brief history of the U.S. Geological Survey's (USGS) streamgaging efforts from the placement of the first USGS gage at Embudo, New Mexico in 1889 to the present day. The site is viewable at https://labs.waterdata.usgs.gov/visualizations/gages-through-the-ages
```

File: src/assets/styles/waterColor.js
Comment (cnell-usgs) [inline src/assets/styles/waterColor.js:1]: Were these files being used somewhere?

File: src/main.js [hunk 5]
```diff
@@ -1,62 +1,23 @@
...
-  var d = new Date().getTime();
-  if (typeof performance !== 'undefined' && typeof performance.now === 'function'){
-    d += performance.now(); //use high-precision timer if available
-  }
-  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
-    var r = (d + Math.random() * 16) % 16 | 0;
-    d = Math.floor(d / 16);
-    return (c === 'x' ? r : (r & 0x3 | 0x8)).toString(16);
-  });
-};
```
Comment (cnell-usgs) [inline src/main.js:54]: huh

**Review by cnell-usgs — APPROVED** [review]
There's a small edit to the capitalization of the georgia slider component needed to build the site, otherwise this looks great.

This is a pretty impressive reduction in the overall repo size. I noticed a lot of files were deleted, seemingly altogether. Were these just unused and extra in the repo?

File: src/components/MapImageSlider.vue (hunk 1 above)
Comment (hcorson-dosch-usgs) [reply src/components/MapImageSlider.vue:1820]: weird - I changed the capitalization in the filename locally, but apparently that didn't update in git

Comment (hcorson-dosch-usgs) [reply src/components/MapImageSlider.vue:1820]: ok I made this edit in GitHub - should now work as is, retaining the component naming convention of starting w/ a capitalized letter

File: src/assets/styles/waterColor.js
Comment (hcorson-dosch-usgs) [reply src/assets/styles/waterColor.js:1]: I'm guessing it was a set of styles explored for use with `src/components/MapboxSlider.vue`, which wasn't used in the final site.

**Comment by hcorson-dosch-usgs** [comment]
> I noticed a lot of files were deleted, seemingly altogether. Were these just unused and extra in the repo?

Yeah, there was quite a bit of unused content, I expect leftover from exploratory work/earlier drafts

File: src/main.js (hunk 5 above)
Comment (hcorson-dosch-usgs) [reply src/main.js:54]: I think this was part of an earlier more elaborate approach to using GA

**Comment by hcorson-dosch-usgs** [comment]
[like]  Corson-Dosch, Hayley R reacted to your message:
________________________________
From: Cee Nell ***@***.***>
Sent: Wednesday, July 17, 2024 6:10:50 PM
To: DOI-USGS/gages-through-the-ages ***@***.***>
Cc: Corson-Dosch, Hayley R ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [DOI-USGS/gages-through-the-ages] Update site to use vite + vue3 (PR #211)




 This email has been received from outside of DOI - Use caution before clicking on links, opening attachments, or responding.



@cnell-usgs commented on this pull request.

________________________________

In src/main.js<https://github.com/DOI-USGS/gages-through-the-ages/pull/211#discussion_r1681504190>:

> -// create a unique ID that can be used to identify user sessions
-// Note: this section is borrowed code with Public Domain/MIT licensing
-const sessionID = function() {
-  var d = new Date().getTime();
-  if (typeof performance !== 'undefined' && typeof performance.now === 'function'){
-    d += performance.now(); //use high-precision timer if available
-  }
-  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
-    var r = (d + Math.random() * 16) % 16 | 0;
-    d = Math.floor(d / 16);
-    return (c === 'x' ? r : (r & 0x3 | 0x8)).toString(16);
-  });
-};


Yeah totally. I think this is built in now, although we don't necessary leverage the individual user data directly, but it is internal to how it's captured

**Comment by hcorson-dosch-usgs** [comment]
[like]  Corson-Dosch, Hayley R reacted to your message:
________________________________
From: Cee Nell ***@***.***>
Sent: Wednesday, July 17, 2024 6:10:50 PM
To: DOI-USGS/gages-through-the-ages ***@***.***>
Cc: Corson-Dosch, Hayley R ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [DOI-USGS/gages-through-the-ages] Update site to use vite + vue3 (PR #211)




 This email has been received from outside of DOI - Use caution before clicking on links, opening attachments, or responding.



@cnell-usgs commented on this pull request.

________________________________

In src/main.js<https://github.com/DOI-USGS/gages-through-the-ages/pull/211#discussion_r1681504190>:

> -// create a unique ID that can be used to identify user sessions
-// Note: this section is borrowed code with Public Domain/MIT licensing
-const sessionID = function() {
-  var d = new Date().getTime();
-  if (typeof performance !== 'undefined' && typeof performance.now === 'function'){
-    d += performance.now(); //use high-precision timer if available
-  }
-  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
-    var r = (d + Math.random() * 16) % 16 | 0;
-    d = Math.floor(d / 16);
-    return (c === 'x' ? r : (r & 0x3 | 0x8)).toString(16);
-  });
-};


Yeah totally. I think this is built in now, although we don't necessary leverage the individual user data directly, but it is internal to how it's captured

**Pull request merged**